### PR TITLE
Lazy initialize feature flags setter.

### PR DESCRIPTION
**Pull request by dblock**
Not sure it's worth it, avoids calling `clear()` on every teardown that doesn't need feature flags, WDYT?

### CODE REVIEW

File: test/framework/src/main/java/org/opensearch/test/FeatureFlagSetter.java [hunk 1]
```diff
@@ -23,22 +23,42 @@
  */
 public class FeatureFlagSetter {
 
+    private static FeatureFlagSetter INSTANCE = new FeatureFlagSetter();
```
Comment (ashking94) [inline test/framework/src/main/java/org/opensearch/test/FeatureFlagSetter.java:26]: shouldn't this be null if we are doing lazy initialisation?

File: test/framework/src/main/java/org/opensearch/test/FeatureFlagSetter.java [hunk 2]
```diff
@@ -23,22 +23,42 @@
...
+
+    public static synchronized FeatureFlagSetter getInstance() {
```
Comment (ashking94) [inline test/framework/src/main/java/org/opensearch/test/FeatureFlagSetter.java:28]: This should be private now.

File: test/framework/src/main/java/org/opensearch/test/FeatureFlagSetter.java [hunk 3]
```diff
@@ -23,22 +23,42 @@
...
+    }
+
     private static final Logger LOGGER = LogManager.getLogger(FeatureFlagSetter.class);
-    private static final Set<String> SET_FLAGS = ConcurrentCollections.newConcurrentSet();
+    private final Set<String> flags = ConcurrentCollections.newConcurrentSet();
 
     @SuppressForbidden(reason = "Enables setting of feature flags")
-    public static void set(String flag) {
-        SET_FLAGS.add(flag);
+    public void setFlag(String flag) {
```
Comment (ashking94) [inline test/framework/src/main/java/org/opensearch/test/FeatureFlagSetter.java:50]: This should be private now too.

File: test/framework/src/main/java/org/opensearch/test/FeatureFlagSetter.java [hunk 4]
```diff
@@ -23,22 +23,42 @@
...
+        flags.add(flag);
         AccessController.doPrivileged((PrivilegedAction<String>) () -> System.setProperty(flag, "true"));
         LOGGER.info("set feature_flag={}", flag);
     }
 
     @SuppressForbidden(reason = "Clears the set feature flags")
-    public static void clearAllFlags() {
-        for (String flag : SET_FLAGS) {
+    public void clearAll() {
```
Comment (ashking94) [inline test/framework/src/main/java/org/opensearch/test/FeatureFlagSetter.java:57]: This should be private now too.

File: test/framework/src/main/java/org/opensearch/test/FeatureFlagSetter.java (hunk 1 above)
Comment (dblock) [reply test/framework/src/main/java/org/opensearch/test/FeatureFlagSetter.java:26]: aha yes

**Comment by ashking94** [comment]
> Not sure it's worth it, avoids calling `clear()` on every teardown that doesn't need feature flags, WDYT?

This is really a better implementation. Have left some comments.

**Comment by dblock** [comment]
Updated, all yours.

**Review by ashking94 — APPROVED** [review]

**Pull request merged**